src/components/HeaderBlock/HeaderBlock.js

```diff
@@ -0,0 +1,343 @@
+            navigation.goBack(null);
+          }}
+          fontSize={fontSizes.extraLarge}
+          horizontalAlign="flex-start"
+        />
+      );
+    }
+
+    return items.map((item) => {
+      if (item.user || item.userIcon) {
+        return this.renderUser(theme, !item.userIcon);
+      }
+      if (item.title) {
+        return (
+          <HeaderTitle theme={theme} key={item.title} style={item.color ? { color: item.color } : {}}>
+            {item.title}
+          </HeaderTitle>
+        );
+      }
+
+      if (item.icon) {
+        return (
+          <ActionIcon
+            key={item.icon}
+            icon={item.icon}
+            color={theme.rightActionIconColor || UIColors.defaultNavigationColor}
+            onPress={item.action}
+            fontSize={fontSizes.extraLarge}
+            horizontalAlign="flex-start"
+          />
+        );
+      }
+      if (item.label) {
+        return (
+          <TextButton onPress={item.onPress} key={item.label}>
+            <ButtonLabel theme={theme}>{item.label}</ButtonLabel>
+          </TextButton>
+        );
+      }
+      if (item.close) {
+        return (
+          <ActionIcon
+            key="close"
+            icon="close"
+            color={baseColors.slateBlack}
+            onPress={() => navigation.goBack()}
+            fontSize={fontSizes.extraSmall}
+            horizontalAlign="flex-start"
+            style={{ marginBottom: -2, marginRight: -4 }}
+          />
+        );
+      }
+      if (item.actionButton) {
+        return (<HeaderActionButton {...item.actionButton} theme={theme} />);
+      }
+      if (item.custom) {
+        return <View key={item.key || 'custom'}>{item.custom}</View>;
+      }
+      return null;
+    });
+  };
+
+  renderUser = (theme, showName: boolean) => {
+    const { user } = this.props;
+    return (
+      <UserButton key="user">
+        <HeaderProfileImage
+          uri={`${user.profileImage}?t=${user.lastUpdateTime || 0}`}
+          userName={user.username}
+          diameter={profileImageWidth}
+          onPress={() => {}}
+          containerStyle={{
+            borderRadius: profileImageWidth / 2,
+            backgroundColor: user.profileImage ? 'transparent' : baseColors.lightGray,
+          }}
+          noShadow
+        />
+        {showName &&
+        <HeaderTitle theme={theme} style={{ marginLeft: spacing.medium }}>{user.username}</HeaderTitle>}
+      </UserButton>
+    );
+  };
+
+  render() {
+    const { floating } = this.props;
+    const theme = getTheme(this.props);
+
+    return (
+      <Wrapper theme={theme} floating={floating}>
+        <SafeArea forceInset={{ bottom: 'never' }} androidStatusbarHeight={StatusBar.currentHeight}>
+          <HeaderContentWrapper>
+            {this.renderHeaderContent(theme)}
+          </HeaderContentWrapper>
+        </SafeArea>
+      </Wrapper>
+    );
+  }
+}
+
+const mapStateToProps = ({
+  user: { data: user },
+}) => ({
+  user,
+});
```

Comment:
I was thinking the same, yet based on new designs there will be some headers with user info on them. So it might save us from repetitive user linking to other screens.